packages/web/src/pages/upload-page/fields/CollectibleGatedFields.tsx

```diff
@@ -0,0 +1,177 @@
+  const [, , { setValue: setAvailabilityValue }] = useField({
+    name: AVAILABILITY_TYPE
+  })
+  const [
+    { value: premiumConditionsValue },
+    ,
+    { setValue: setPremiumConditionsValue }
+  ] =
+    useField<TrackAvailabilityFormValues[typeof PREMIUM_CONDITIONS]>(
+      PREMIUM_CONDITIONS
+    )
+
+  const { ethCollectionMap, solCollectionMap } = useSelector(
+    getSupportedUserCollections
+  )
+  const hasUnsupportedCollection = useSelector(getHasUnsupportedCollection)
+
+  const ethCollectibleItems = useMemo(() => {
+    return Object.keys(ethCollectionMap)
+      .sort((s1, s2) =>
+        ethCollectionMap[s1].name.localeCompare(ethCollectionMap[s2].name)
+      )
+      .map((slug) => ({
+        text: ethCollectionMap[slug].name,
+        el: (
+          <div className={styles.dropdownRow}>
+            {!!ethCollectionMap[slug].img && (
+              <img
+                src={ethCollectionMap[slug].img!}
+                alt={ethCollectionMap[slug].name}
+              />
+            )}
+            <span>{ethCollectionMap[slug].name}</span>
+          </div>
+        ),
+        value: slug
+      }))
+  }, [ethCollectionMap])
+
+  const solCollectibleItems = useMemo(() => {
+    return Object.keys(solCollectionMap)
+      .sort((m1, m2) =>
+        solCollectionMap[m1].name.localeCompare(solCollectionMap[m2].name)
+      )
+      .map((mint) => ({
+        text: solCollectionMap[mint].name,
+        el: (
+          <div className={styles.dropdownRow}>
+            {!!solCollectionMap[mint].img && (
+              <img
+                src={solCollectionMap[mint].img!}
+                alt={solCollectionMap[mint].name}
+              />
+            )}
+            <span>{solCollectionMap[mint].name}</span>
+          </div>
+        ),
+        value: mint
+      }))
+  }, [solCollectionMap])
+
+  const menuItems = useMemo(
+    () => [...ethCollectibleItems, ...solCollectibleItems],
+    [ethCollectibleItems, solCollectibleItems]
+  )
+
+  // If no nft collection was previously selected, then the default value is an empty string,
+  // which makes the dropdown show the placeholder.
+  // Otherwise, the default value is the nft collection which was previously selected,
+  // which also includes the collection image.
+  const defaultCollectionName =
+    premiumConditionsValue?.nft_collection?.name ?? ''
+  const defaultCollection = menuItems.find(
+    (item) => item.text === defaultCollectionName
+  )
+  const defaultValue = defaultCollection || defaultCollectionName
+
+  const renderFooter = useCallback(() => {
+    return hasUnsupportedCollection ? (
+      <HelpCallout
+        className={styles.helpCallout}
+        content={
+          <div>
+            <div>{messages.compatibilityTitle}</div>
+            <div>{messages.compatibilitySubtitle}</div>
+          </div>
+        }
+      />
+    ) : null
+  }, [hasUnsupportedCollection])
+
+  return (
+    <div className={styles.root}>
+      <DropdownInput
```

Comment:
Probably need to replace this with a field now that I think of it. Haven't tested this part yet.

Reply:
The onSelect below might be good enough, but I think I'll need to change the defaultValue at least

Reply:
yeah ideally we have the input wrapped in the formik setup :) before the end of this epic id like to see us have a canonical `DropdownField`